server/src/main/java/org/apache/cloudstack/diagnostics/DiagnosticsDataObject.java

```diff
@@ -0,0 +1,100 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package org.apache.cloudstack.diagnostics;
+
+import com.cloud.agent.api.Answer;
+import com.cloud.agent.api.to.DataObjectType;
+import com.cloud.agent.api.to.DataTO;
+import org.apache.cloudstack.engine.subsystem.api.storage.DataObject;
+import org.apache.cloudstack.engine.subsystem.api.storage.DataStore;
+import org.apache.cloudstack.engine.subsystem.api.storage.ObjectInDataStoreStateMachine;
+
+public class DiagnosticsDataObject implements DataObject {
+    private DataTO dataTO;
+    private DataStore dataStore;
+
+    public DiagnosticsDataObject(DataTO dataTO, DataStore dataStore) {
+        this.dataTO = dataTO;
+        this.dataStore = dataStore;
+    }
+
+    @Override
+    public long getId() {
+        return 0;
```

Comment:
Why not normal getters? If parameters are not set then the default value is retrieved on each getter

Reply:
Changed to return parameter